presto/eval/treesat_eval.py

```diff
@@ -347,51 +547,38 @@ def finetuning_results(
+                models=sklearn_modes,
+            )
+            results_dict.update(
+                self.evaluate_for_sklearn(sklearn_model_dict, pretrained_model, mask)
             )
-            return self.evaluate(sklearn_model_dict, pretrained_model, mask)
-        return {}
-
-    def finetune(self, pretrained_model, mask: Optional[np.ndarray] = None) -> FineTuningModel:
-        raise NotImplementedError
-        # hyperparams = Hyperparams()
-        # prop_val_images = 0.1
-        # model = self._construct_finetuning_model(pretrained_model)
-        #
-        # opt = Adam(model.parameters(), lr=hyperparams.lr)
-        # loss_fn = nn.BCEWithLogitsLoss(reduction="mean")
-        #
-        # x, dw, latlons, target, image_names = self.load_npys(test=False)
-        #
-        # unique_image_names = np.unique(image_names)
-        # num = round(prop_val_images * len(unique_image_names))
-        # val_images = np.random.choice(unique_image_names, size=num, replace=False)
-        # val_filter = np.isin(image_names, val_images)
-        #
-        # train_dl = DataLoader(
-        #     TensorDataset(
-        #         torch.from_numpy(x[~val_filter]).to(device).float(),
-        #         torch.from_numpy(dw[~val_filter]).to(device).long(),
-        #         torch.from_numpy(latlons[~val_filter]).to(device).float(),
-        #         torch.from_numpy(target[~val_filter]).to(device).float(),
-        #         torch.full(target[~val_filter].shape[:1], self.start_month, device=device).long(),
-        #     ),
-        #     batch_size=hyperparams.batch_size,
-        #     shuffle=True,
-        # )
-        #
-        # val_dl = DataLoader(
-        #     TensorDataset(
-        #         torch.from_numpy(x[val_filter]).to(device).float(),
-        #         torch.from_numpy(dw[val_filter]).to(device).long(),
-        #         torch.from_numpy(latlons[val_filter]).to(device).float(),
-        #         torch.from_numpy(target[val_filter]).to(device).float(),
-        #         torch.full(target[val_filter].shape[:1], self.start_month, device=device).long(),
-        #     ),
-        #     batch_size=hyperparams.batch_size,
-        #     shuffle=False,
-        # )
-        #
-        # return self.finetune_pytorch_model(
-        #     model, hyperparams, opt, train_dl, val_dl, loss_fn, loss_fn, mask
-        # )
+        return results_dict
+
+    def finetune_with_aggregate(
+        self, pretrained_model, mask: Optional[np.ndarray], aggregate: str
+    ) -> FineTuningModel:
+        hparams = Hyperparams(batch_size=self.batch_size)
+        model = self._construct_finetuning_model_with_aggregates(pretrained_model, aggregate)
+        opt = AdamW(model.parameters(), lr=hparams.lr, weight_decay=hparams.weight_decay)
+
+        loss_fn = nn.CrossEntropyLoss(reduction="mean")
+        ds = TreeSatDataset(self.input_patch_size, split="train")
+        train_ds, val_ds = ds.train_val_split(val_ratio=0.1, seed=self.seed)
```

Comment:
This means you'll end up with a different validation set per seed?

Reply:
Yes, but not different test sets.

Reply:
I get that this is another way to use seeds to 'cancel' random effects during training, but from the perspective of making experiments comparable, it seems more practical to use the same validation set? + other tasks use identical val sets for different seeds too?